receivers-in-go.md

```diff
@@ -0,0 +1,257 @@
+called on an instance, but otherwise there is very little magic going on.
+
+In other languages where `this` and `self` is a thing (Python, Ruby, JavaScript, 
+and so on) it's a much more complicated situation. These are not vanilla local 
+variables wearing fancy pants. The thing we might expect `this` to represent 
+inside of a method could actually represent something very different once 
+inheretance or metaclasses had their way. In effect, it might not make any sense 
+to give contextual names to `self` in Python, but it definitely makes sense in 
+Go.
+
+## Reshaping our code
+
+As we start refactoring our code, we take a chunk of code that is already 
+functional and put it in another context where it allows for more flexibility 
+towards a higher-level goal.
+
+The first step of taking flat code and embedding it into a struct method is 
+usually fine regardless of what the receiver is named. At least in the 
+beginning.
+
+Later, once we start moving pieces of code *between* levels abstractions or from 
+higher levels of abstraction into a lower level, this is when consistently 
+well-named receivers make a huge difference.
+
+Imagine taking this snippet from a higher-level container like `Room` which 
+holds groups of users in a server, and moving it up or down one level:
+
+```go
+func (room *Room) Announce() {
+    srv := room.Server()
+    for _, c := range srv.Clients() {
+        // Send announcement to all clients about a new room
+        c.Send(srv.RenderAnnouncement(room))
+    }
+}
+
+// Moved between...
+
+func (srv *Server) AddRoom(room *Room) {
+    for _, c := range srv.Clients() {
+        // Send announcement to all clients about a new room
+        c.Send(srv.RenderAnnouncement(room))
+    }
+}
+```
+
+This is a great little pattern to keep everything working despite moving between 
+layers of abstraction. Note how the inner code stays identical and all we're 
+doing is adding a little extra context outside of it.
+
+On the other hand:
+
+```go
+func (this *Room) Announce() {
+    srv := this.Server()
+    for _, c := range srv.Clients() {
+        // Send announcement to all clients about a new room
+        c.Send(srv.RenderAnnouncement(this))
+    }
+}
+
+// Moved between...
+
+func (this *Server) AddRoom(room *Room) {
+    for _, c := range this.Clients() {
+        // Send announcement to all clients about a new room
+        c.Send(this.RenderAnnouncement(room))
+    }
+}
+```
+
+When using `this`, there is confusion about whether we're referring to the 
+server or the room as we're moving the code between.
+
+```diff
+-       c.Send(this.RenderAnnouncement(room))
++       c.Send(srv.RenderAnnouncement(this))
+```
+
+Yes, this is going to cause some bugs that hopefully the compiler catches (or 
+maybe not, if the interfaces happen to be compatible). More importantly, it 
+makes moving code around more tedious.
+
+
+## Should receivers be special?
```

Comment:
This heading kinda throws me off - I think this section works best as a conclusion for everything above. For the sake of flow, it may work to remove the header.